EventGenerator/src/GenFilter_module.cc

```diff
@@ -75,42 +109,70 @@ namespace mu2e {
 
   bool GenFilter::filter(art::Event& event) {
     if(isNull_) return true;
-    bool passed = false;
+    if(!(maxr_cut_ || energy_cut_ || momentum_cut_ || pT_cut_)) return true;
     auto sim = event.getValidHandle<StageParticleCollection>(_SimToken);
     _SimCol = sim.product();
+    bool pass_maxr_cut = !maxr_cut_;
+    bool pass_energy_cut = !energy_cut_;
+    bool pass_momentum_cut = !momentum_cut_;
+    bool pass_pT_cut = !momentum_cut_;
     for(const auto& aParticle : *_SimCol){
-    //  make momentum and position vectors
+      //  make momentum and position vectors
       GeomHandle<DetectorSystem> det;
       ROOT::Math::XYZVectorF pos = XYZVectorF(det->toDetector(aParticle.position()));
       ROOT::Math::XYZTVector pos0(pos.x(), pos.y(), pos.z(), aParticle.time());
       ROOT::Math::PxPyPzMVector mom0(aParticle.momentum().x(), aParticle.momentum().y(), aParticle.momentum().z(), aParticle.momentum().t());
+      _energy = aParticle.momentum().e();
+      _mom = aParticle.momentum().rho();
+      _momT = sqrt(mom0.x()*mom0.x() + mom0.y()*mom0.y());
 
-      // extract charge
-      static GlobalConstantsHandle<ParticleDataList> pdt;
-      auto charge = pdt->particle(aParticle.pdgId()).charge();
+      if(maxr_cut_) { // cut on maximum R in the solenoid
+        // extract charge
+        static GlobalConstantsHandle<ParticleDataList> pdt;
+        auto charge = pdt->particle(aParticle.pdgId()).charge();
 
-      // extact field
-      GeomHandle<BFieldManager> bfmgr;
-      mu2e::GeomHandle<mu2e::Tracker> tracker;
-      auto tracker_origin = det->toMu2e(tracker->origin());
-      //XYZVectorF pos3Vec = XYZVectorF(aParticle.position().x(),aParticle.position().y(),aParticle.position().z());
-      ROOT::Math::XYZVector bnom(bfmgr->getBField(tracker_origin));
+        // check if the particle is charged, else ignore this cut
+        if(fabs(charge) > 1.e-5) {
+          // extact field
+          GeomHandle<BFieldManager> bfmgr;
+          mu2e::GeomHandle<mu2e::Tracker> tracker;
+          auto tracker_origin = det->toMu2e(tracker->origin());
+          //XYZVectorF pos3Vec = XYZVectorF(aParticle.position().x(),aParticle.position().y(),aParticle.position().z());
+          ROOT::Math::XYZVector bnom(bfmgr->getBField(tracker_origin));
+
+          // make the loophelix
+          KinKal::LoopHelix lh(pos0, mom0, charge, bnom);
+          // calculate rmax and add maxr to siminfo
+          _maxr = sqrt(lh.cx()*lh.cx()+lh.cy()*lh.cy())+fabs(lh.rad());
+          pass_maxr_cut |= _maxr < maxr_max_ && _maxr > maxr_min_;
```

Comment:
This was written with IPA DIOs in mind I believe, filtering out DIO events that can't make it to the tracker volume. For the conversions, I only use the energy cuts as of now to veto more symmetric conversions. As we'll generate E_gamma > 85 MeV, there's no reason to be saving tracks with E < 84 MeV since we've omitted a relevant part of the photon spectrum already. For RMC as a general background/pileup this cut likely shouldn't be applied, but it's useful towards targeting the high momentum e+/e- component of the background.

Reply:
By default, each filter option is off. The only change that I had to propagate to current Production fcl is to add `filterMaxR: true` to turn on this filtering, and I set `isNull` to default to false since each individual filter option defaults to false. Turning off all filtering can still be achieved by setting `isNull : true` as before.

Reply:
Yes, as long as the general default is "no filtering" since this is in the path for all other production primaries too

Reply:
Yes, the default is still no filtering so other primary production paths should be unaffected by these changes.

Reply:
Having all these cuts in a single module raises the possibility they might get used in ways that cause physics biases. For now we can deal with this in production, but before this filter expands again lets think through the design.